numba/targets/arrayobj.py

```diff
@@ -1491,6 +1491,8 @@ def array_transpose_vararg(context, builder, sig, args):
 
 @overload(np.transpose)
 def numpy_transpose(a, axes=None):
+    if isinstance(a, types.BaseTuple):
+        raise errors.TypingError("np.transpose does not accept tuples")
```

Comment:
Should probably raise an `error.UnsupportedError` ? It's not that it cannot be typed/it's bad, it's just Numba won't handle it.

Reply:
check